clients/rust/src/generated/errors/mpl_token_metadata.rs

```diff
@@ -0,0 +1,598 @@
+//! This code was AUTOGENERATED using the kinobi library.
+//! Please DO NOT EDIT THIS FILE, instead use visitors
+//! to add features, then rerun kinobi to update it.
+//!
+//! [https://github.com/metaplex-foundation/kinobi]
+//!
+
+use num_derive::FromPrimitive;
+use thiserror::Error;
+
+#[derive(Clone, Debug, Eq, Error, FromPrimitive, PartialEq)]
+pub enum MplTokenMetadataError {
+    /// 0x0 -
+    #[error("")]
+    InstructionUnpackError,
+    /// 0x1 -
+    #[error("")]
+    InstructionPackError,
+    /// 0x2 - Lamport balance below rent-exempt threshold
+    #[error("Lamport balance below rent-exempt threshold")]
+    NotRentExempt,
+    /// 0x3 - Already initialized
+    #[error("Already initialized")]
+    AlreadyInitialized,
+    /// 0x4 - Uninitialized
+    #[error("Uninitialized")]
+    Uninitialized,
+    /// 0x5 -  Metadata's key must match seed of ['metadata', program id, mint] provided
+    #[error(" Metadata's key must match seed of ['metadata', program id, mint] provided")]
+    InvalidMetadataKey,
+    /// 0x6 - Edition's key must match seed of ['metadata', program id, name, 'edition'] provided
+    #[error("Edition's key must match seed of ['metadata', program id, name, 'edition'] provided")]
+    InvalidEditionKey,
+    /// 0x7 - Update Authority given does not match
+    #[error("Update Authority given does not match")]
+    UpdateAuthorityIncorrect,
+    /// 0x8 - Update Authority needs to be signer to update metadata
+    #[error("Update Authority needs to be signer to update metadata")]
+    UpdateAuthorityIsNotSigner,
+    /// 0x9 - You must be the mint authority and signer on this transaction
+    #[error("You must be the mint authority and signer on this transaction")]
+    NotMintAuthority,
+    /// 0xA - Mint authority provided does not match the authority on the mint
+    #[error("Mint authority provided does not match the authority on the mint")]
+    InvalidMintAuthority,
+    /// 0xB - Name too long
+    #[error("Name too long")]
+    NameTooLong,
+    /// 0xC - Symbol too long
+    #[error("Symbol too long")]
+    SymbolTooLong,
+    /// 0xD - URI too long
+    #[error("URI too long")]
+    UriTooLong,
+    /// 0xE -
+    #[error("")]
+    UpdateAuthorityMustBeEqualToMetadataAuthorityAndSigner,
+    /// 0xF - Mint given does not match mint on Metadata
+    #[error("Mint given does not match mint on Metadata")]
+    MintMismatch,
+    /// 0x10 - Editions must have exactly one token
+    #[error("Editions must have exactly one token")]
+    EditionsMustHaveExactlyOneToken,
+    /// 0x11 -
+    #[error("")]
+    MaxEditionsMintedAlready,
+    /// 0x12 -
+    #[error("")]
+    TokenMintToFailed,
+    /// 0x13 -
+    #[error("")]
+    MasterRecordMismatch,
+    /// 0x14 -
+    #[error("")]
+    DestinationMintMismatch,
+    /// 0x15 -
+    #[error("")]
+    EditionAlreadyMinted,
+    /// 0x16 -
+    #[error("")]
+    PrintingMintDecimalsShouldBeZero,
```

Comment:
Nitpick: I know we removed some of these error messages because of program space but we could add them back on the Kinobi config. In fact, we could remove even more on the program to clear more space and add them back on the clients.

Something to consider for a separate PR perhaps.

Reply:
I like this. we can add the ones that are missing now.